Fix PlayStation build following 278442@main

rkirsling opened the pull request:
#### 7bf250e4832adfb28c2bf2bc518ad59958f2ae1b
<pre>
Fix PlayStation build following 278442@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=273860">https://bugs.webkit.org/show_bug.cgi?id=273860</a>

Reviewed by Chris Dumez.

PS needs to implement std::u8string, which itself would be trivial,
but the complex part is that the return type of std::filesystem::u8string changed from std::string to std::u8string.
We already have accommodation for this in WebKit, so let&apos;s just extend it a little further for now.

* Source/WTF/wtf/FileSystem.cpp:
(WTF::FileSystemImpl::toStdFileSystemPath):
(WTF::FileSystemImpl::fromStdFileSystemPath): Use span8 when HAVE(MISSING_U8STRING).
* Source/WTF/wtf/PlatformHave.h:
Rename HAVE(MISSING_STD_FILESYSTEM_PATH_CONSTRUCTOR) to HAVE(MISSING_U8STRING), which is what it actually means anyway.

* Source/WTF/wtf/text/StringCommon.h:
Guard overload with !HAVE(MISSING_U8STRING).

Canonical link: <a href="https://commits.webkit.org/278490@main">https://commits.webkit.org/278490@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ad54f326335539726b31ae35d3cc0aebd0ee76d3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50744 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30041 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3061 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54003 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1435 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36304 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1085 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/41352 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52843 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27687 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43712 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22477 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/25052 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/968 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9185 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/44080 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47033 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1030 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55593 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/50247 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25846 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/920 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/27103 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/43822 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47828 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/11113 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27971 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/57722 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7345 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/26835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11884 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WTF/wtf/text/StringCommon.h:
```diff
@@ -58,7 +58,11 @@ inline std::span<const char> span(const char* string)
 inline std::span<const char8_t> span(const std::u8string& string)
+#endif
 {
     return { string.data(), string.length() };
```
doesn't `string.data()` give you a `const char*`? If so, how can you construct a `std::span<const char8_t>` from it? Is `char8_t` the same thing as `char` on Playstation?

rkirsling replied:
Whoops, thanks for noticing that. Was trying to simplify but went too far—I'll just use `span8()` instead.

rkirsling committed:
Fix PlayStation build following 278442@main
https://bugs.webkit.org/show_bug.cgi?id=273860

Reviewed by Chris Dumez.

PS needs to implement std::u8string, which itself would be trivial,
but the complex part is that the return type of std::filesystem::u8string changed from std::string to std::u8string.
We already have accommodation for this in WebKit, so let's just extend it a little further for now.

* Source/WTF/wtf/FileSystem.cpp:
(WTF::FileSystemImpl::toStdFileSystemPath):
(WTF::FileSystemImpl::fromStdFileSystemPath): Use span8 when HAVE(MISSING_U8STRING).
* Source/WTF/wtf/PlatformHave.h:
Rename HAVE(MISSING_STD_FILESYSTEM_PATH_CONSTRUCTOR) to HAVE(MISSING_U8STRING), which is what it actually means anyway.

* Source/WTF/wtf/text/StringCommon.h:
Guard overload with !HAVE(MISSING_U8STRING).

Canonical link: https://commits.webkit.org/278490@main
